File: airlines-gds-backend/services/flight-service/src/main/java/com/gds/airline/flight_service/repository/FlightRepository.java
package com.gds.airline.flight_service.repository;

import com.gds.airline.flight_service.model.Flight;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface FlightRepository extends JpaRepository<Flight, Long> {
    // Spring Data crea la consulta SQL automáticamente solo con leer el nombre del método
    List<Flight> findByOriginAndDestination(String origin, String destination);
}

File: skylink-mobileApp/app/src/main/java/com/alalodev/skylink/features/home/domain/repository/FlightRepository.kt
package com.alalodev.skylink.features.home.domain.repository

import com.alalodev.skylink.core.network.util.NetworkResult
import com.alalodev.skylink.features.home.data.model.Flight

interface FlightRepository {
    suspend fun searchFlights(origin: String, destination: String): NetworkResult<List<Flight>>
    suspend fun getAllFlights(): NetworkResult<List<Flight>>
}
